File: app/src/main/java/pl/pola_app/model/PhotoPicturesReport.java
package pl.pola_app.model;


import com.google.gson.annotations.SerializedName;

public class PhotoPicturesReport {

    @SerializedName("product_id")
    int productId;
    @SerializedName("files_count")
    int filesCount;
    @SerializedName("file_ext")
    String fileExt;
    @SerializedName("mime_type")
    String mineType;
    @SerializedName("original_width")
    int originalWidth;
    @SerializedName("original_height")
    int originalHeight;
    @SerializedName("width")
    int width;
    @SerializedName("height")
    int height;
    @SerializedName("device_name")
    String deviceName;
    @SerializedName("flash_used")
    boolean flashUsed;
    @SerializedName("was_portrait")
    boolean wasPortrait;

    public static class Builder {
        int productId;
        int filesCount;
        int originalWidth;
        int originalHeight;
        int width;
        int height;
        String deviceName;
        protected Builder() {

        }

        public Builder setProductId(int productId) {
            this.productId = productId;
            return  this;
        }

        public Builder setFilesCount(int filesCount) {
            this.filesCount = filesCount;
            return  this;
        }

        public Builder setOriginalWidth(int originalWidth) {
            this.originalWidth = originalWidth;
            return  this;
        }

        public Builder setOriginalHeight(int originalHeight) {
            this.originalHeight = originalHeight;
            return  this;
        }

        public Builder setDeviceName(String deviceName) {
            this.deviceName = deviceName;
            return  this;
        }

        public Builder setWidth(int width) {
            this.width = width;
            return this;
        }

        public Builder setHeight(int height) {
            this.height = height;
            return this;
        }

        public PhotoPicturesReport build(){
            PhotoPicturesReport result = new PhotoPicturesReport();
            result.productId = productId;
            result.filesCount = filesCount;
            result.fileExt = "jpg";
            result.mineType = "image/*";
            result.originalWidth = originalWidth;
            result.originalHeight = originalHeight;
            result.width = width;
            result.height = height;
            result.deviceName = deviceName;
            result.flashUsed = false;
            result.wasPortrait = true;
            return  result;
        }
    }

    public static Builder builder() {
        return new Builder();
    }
}

File: app/src/main/java/pl/pola_app/model/PhotoPicturesReport.kt
package pl.pola_app.model

import com.google.gson.annotations.SerializedName

data class PhotoPicturesReport private constructor(
    @SerializedName("product_id")
    val productId: Int = 0,

    @SerializedName("files_count")
    val filesCount: Int = 0,

    @SerializedName("file_ext")
    val fileExt: String,

    @SerializedName("mime_type")
    val mineType: String,

    @SerializedName("original_width")
    val originalWidth: Int = 0,

    @SerializedName("original_height")
    val originalHeight: Int = 0,

    @SerializedName("width")
    val width: Int = 0,

    @SerializedName("height")
    val height: Int = 0,

    @SerializedName("device_name")
    val deviceName: String,

    @SerializedName("flash_used")
    val flashUsed: Boolean = false,

    @SerializedName("was_portrait")
    val wasPortrait: Boolean = false
){


    class Builder {
        var productId = 0
            private set
        var filesCount = 0
            private set
        var originalWidth = 0
            private set
        var originalHeight = 0
            private set
        var width = 0
            private set
        var height = 0
            private set
        var deviceName: String = ""
            private set

        fun productId(productId: Int) = apply { this.productId = productId }
        fun filesCount(filesCount: Int) = apply { this.filesCount = filesCount }
        fun originalWidth(originalWidth: Int) = apply { this.originalWidth = originalWidth }
        fun originalHeight(originalHeight: Int) = apply { this.originalHeight = originalHeight }
        fun width(width: Int) = apply { this.width = width }
        fun height(height: Int) = apply { this.height = height }
        fun deviceName(deviceName: String) = apply { this.deviceName = deviceName }

        fun build(): PhotoPicturesReport {
            return PhotoPicturesReport(
                productId,
                filesCount,
                fileExt = "jpg",
                mineType = "image/*",
                originalWidth,
                originalHeight,
                width,
                height,
                deviceName,
                flashUsed = false,
                wasPortrait = true
            )
        }
    }

    companion object {
        fun builder(): Builder {
            return Builder()
        }
    }
}